pkg/services/publicdashboards/service/service.go

```diff
@@ -428,9 +428,21 @@ func GenerateAccessToken() (string, error) {
 }
 
 func (pd *PublicDashboardServiceImpl) newCreatePublicDashboard(ctx context.Context, dto *SavePublicDashboardDTO) (*PublicDashboard, error) {
-	uid, err := pd.NewPublicDashboardUid(ctx)
-	if err != nil {
-		return nil, err
+	var uid string
+	var err error
+	//Check if uid already exists, if none then auto generate
+	existingPubdash, _ := pd.store.Find(ctx, dto.PublicDashboard.Uid)
+	if existingPubdash != nil {
+		return nil, ErrPublicDashboardUidExists.Errorf("Create: public dashboard uid %s already exists", dto.Uid)
+	}
+
+	if dto.PublicDashboard.Uid != "" {
+		uid = dto.PublicDashboard.Uid
+	} else {
+		uid, err = pd.NewPublicDashboardUid(ctx)
+		if err != nil {
+			return nil, err
+		}
 	}
```

Comment:
```suggestion
    uid :=  dto.PublicDashboard.Uid
	if dto.PublicDashboard.Uid == "" {
		uid, err = pd.NewPublicDashboardUid(ctx)
		if err != nil {
			return nil, err
		}
	}
```

I would already assign the uid to be dto.PublicDashboard.Uid by default, so there's no need to make an if else